backends/eopkg/eopkgBackend.py

```
@@ -692,32 +770,51 @@ def download_packages(self, directory, package_ids):
+                    metadata.package.architecture,
+                    "local",
+                )
                 self.package(pkg_id, INFO_INSTALL, metadata.package.summary)
                 for dep in metadata.package.runtimeDependencies():
                     if not dep.satisfied_by_installed():
                         if not self.packagedb.has_package(dep.package):
-                            self.error(ERROR_DEP_RESOLUTION_FAILED, "Cannot install: %s. Can't resolve dependency %s" % (f, dep.package))
+                            self.error(
+                                ERROR_DEP_RESOLUTION_FAILED,
+                                "Cannot install: %s. Can't resolve dependency %s"
+                                % (f, dep.package),
```

Ruff (0.15.12)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)